### ISSUE TITLE
Handling external calls to note:// URL scheme

### ISSUE THREAD

**Issue by fmakowski** [pbek/QOwnNotes#602]
Hello there,

I want to propose an enhancement, that would allow QOwnNotes to open notes by using URL from external sources. 

For example, internal-like link `note://name-of-my-note-here` would open up the note that can be identified by this URL, or else it would prompt `Sorry, we couldn't find the note with that URL`.

The reason for that is that often, it is very useful to make an annotation somewhere and be able to easily go to that note - my usecase is Taskwarrior, where I would rather store the whole knowledge in the note, but be able to quickly go there from shell when going through my task list.

The issues to be discussed are as follows:

* in issue #266 the `note://` protocol was considered obsolete or soon-deprecated, is that so or maybe are you willing to use it internally, and only deprecate it only from the user perspective?
* I have no clue about cross-platform solution for that - I have only investigated the idea on Linux, and adding `MimeType=x-scheme-handler/note` in `QOwnNotes.desktop` and `x-scheme-handler/note=QOwnNotes.desktop` in `mimetype.list` was enough to fire up the app when `note://` was used externally.
* I see that there is no command line parsing used yet, so as far as I understand arguments passing in that case, it should be implemented.
* when using the method mentioned above, new instance of QOwnNotes is started, instead of using the one that is already running - not yet thought about the solution.
* obviously, when the filenames change, the external link will be unusable. Maybe introduce internal note UUIDs? (or maybe they are already there?)

_I am willing to implement the proposal by myself_, but first, I need some insight on what you think about that (do you find it useful? maybe some additional things to cover?), and also, is there are any development conflicts in the near future.

Best regards!
Filip

**Comment by pbek** [comment]
Thank you for your interest, but I think I still can't follow you what you are proposing. :)

**Comment by fmakowski** [comment]
Let me rephrase it:

Imagine having some other place, where you store your data (website, shell application, calendar, somewhere). Now, you add a link there, which calls "note://my-note-in-qownnotes". This would allow useful linking to internal notes in your QOwnNotes, using similar protocol as the one used internally by QOwnNotes.

When you click on it right now, your browser will not understand what there is to be done, since "note://" protocol is not registered. But adding some scheme handling, plus adding argument parsing on the QOwnNotes side, will result in the feature described above.

Hopefully I made it clearer a little bit ;)

**Comment by pbek** [comment]
Frankly I'm still not sure what you want to be happening. You want to get rid of `note://` links? I haven't found an other solution yet, that works with the syntax highlighting, link opening in the note edit and in the preview. Everything I was trying in #266 was causing a lot of troubles.

There is no data stored about note files (and isn't supposed to), QOwnNotes takes just the note files and generates an in-memory database from them.

**Comment by pbek** [comment]
...so we can't use a UUID or something.
I'm not sure if media handler of any kind will work platform independently and QOwnNotes needs the note to be inside the current note folder.

**Comment by fmakowski** [comment]
About getting rid of the note:// : actually, just the opposite ;) I want to
extend the usability of note:// to external apps, so notes can be opened in
QOwnNotes by clicking in, for example, the link in the browser.

02.06.2017 9:42 PM "Patrizio Bekerle" <notifications@github.com> napisał(a):

Frankly I'm still not sure what you want to be happening. You want to get
rid of note:// links? I haven't found an other solution yet, that works
with the syntax highlighting, link opening in the note edit and in the
preview. Everything I was trying in #266
<https://github.com/pbek/QOwnNotes/issues/266> was causing a lot of
troubles.

There is no data stored about note files (and isn't supposed to), QOwnNotes
takes just the note files and generates an in-memory database from them.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/pbek/QOwnNotes/issues/602#issuecomment-305891169>, or mute
the thread

**Comment by pbek** [comment]
Well then, I see a lot of obstacles for you to overcome, but you seem to like challenges. :)

**Comment by fmakowski** [comment]
Ok, so I will try to work on that a little, we will see how it goes. At some point I will possibly show you some progress so you can see it for yourself if it makes sense, and then we'll consider if it should be pulled back to origin or not. 

Best regards!

**Comment by pbek** [comment]
I'm thrilled. :) Good luck and have fun!

**Comment by pbek** [comment]
Could you please head over to #266 to test and comment on the new features.

**Comment by pbek** [comment]
Meanwhile there also is a special highlighting in the editor if the linked note doesn't exist, so I'm closing this for now.
